Content.Server/DeviceNetwork/Systems/NetworkConfiguratorSystem.cs

```diff
@@ -148,17 +148,19 @@ private void TryLinkDevice(EntityUid uid, NetworkConfiguratorComponent configura
             return;
         }
 
-        if (HasComp<DeviceLinkSourceComponent>(target) && HasComp<DeviceLinkSourceComponent>(configurator.ActiveDeviceLink)
-            || HasComp<DeviceLinkSinkComponent>(target) && HasComp<DeviceLinkSinkComponent>(configurator.ActiveDeviceLink))
+        if (configurator.ActiveDeviceLink.HasValue
+            && (HasComp<DeviceLinkSourceComponent>(target)
+            && HasComp<DeviceLinkSinkComponent>(configurator.ActiveDeviceLink)
+            || HasComp<DeviceLinkSinkComponent>(target)
+            && HasComp<DeviceLinkSourceComponent>(configurator.ActiveDeviceLink)))
```

Comment:
Don't you have to put the logic after `configurator.ActiveDeviceLink.HasValue` in inbetween `()` so the `configurator.ActiveDeviceLink.HasValue` applies to both sides of the or expression?

Reply:
Logic after the  `configurator.ActiveDeviceLink.HasValue` is already in parentheses, if we simplify all arguments to letters and make it one line its more clear: (a && (b && c || d && e))

Reply:
sorry I'm blind apparently and didn't see the parentheses (I was on my phone)